internal/pool/pool.go

```diff
@@ -42,6 +42,9 @@ type (
 		Cap() int
 		// Reset deletes all items currently in the pool and resets the allocated count.
 		Reset()
+		// Range calls f sequentially for each key and value present in the map.
+		// If f returns false, range stops the iteration.
+		Range(f func(item T) bool)
```

Comment:
Perhaps it would be more natural to add a function that get's called when item is removed.

Reply:
I followed https://pkg.go.dev/sync#Map.Range. Would you mind sharing how that would work? I'm looking for a mechanism to clear all the remaining items in the pool when shutting down so we don't end up with dangling SFTP sessions when tearing down the SSH connection.